Js embed modern frameworks

ggrossetie opened the pull request:
I can't push on your branch so I've created a new pull request based on your work with two fixes.

ping @obilodeau

ggrossetie committed:
Asciidoctor-reveal.js integration with Babel Test Case

obilodeau committed:
Instructions to load a devel version for testing

jirutka committed:
Ported adoc-html5s javascript improvements into here

Specifically this commit:
https://github.com/jirutka/asciidoctor-html5s/commit/4702cb2602250576f6cc956c3b8108dc41989d07

With tweaks by me.

obilodeau committed:
Got further in making tests pass but still not

ggrossetie committed:
Fix the converter registration

ggrossetie committed:
Check the converter version

Mogztter commented on test/smoke.js:
```diff
@@ -1,19 +1,25 @@
+const result = asciidoctor.convert(content, options)
 
-const result = asciidoctor.convert(content, options);
+expect(asciidoctorRevealjs.getVersion()).to.be('1.1.4-dev')
```
We should probably use a variable, otherwise we will have to update this value when we release a new version :disappointed:

obilodeau replied:
Sourced it out of `package.json` in 289e390. I think it should work in all cases.

Mogztter commented on src/asciidoctor-revealjs.tmpl.js:
```diff
@@ -0,0 +1,46 @@
+   * Registers the reveal.js converter.
+   */
+  function register () {
+    return resolveModule()
```
The function resolves the module (ie. initialize the module only once).

obilodeau replied:
I'm not the one you should ask that unfortunately :(

Mogztter replied:
@obilodeau I think it's fine because it can be useful 😉

Mogztter commented on lib/asciidoctor-revealjs.rb:
```diff
@@ -1,5 +1,6 @@
 if RUBY_ENGINE == 'opal'
   require 'asciidoctor-revealjs/converter'
+  require 'asciidoctor-revealjs/version'
```
`version` is required so we can get the version of the converter from the API.

```js
const revealjsConverter = require('asciidoctor-reveal.js')
console.log(revealjsConverter.getVersion()) // 1.1.4-dev
```

obilodeau committed:
Build step of the node package is now done with rake

- it simplified node dependencies
- adjusted documentation
- adjusted Travis build

obilodeau committed:
node: Adjusted examples generation to use the new registration mechanism

obilodeau committed:
node: usage doc update (register the back-end)

obilodeau committed:
node: make version test dynamic based on package.json value

obilodeau committed:
Rake fix: now that we build the node package, create the dist/ dir too

obilodeau commented on test/js-babel/README.adoc:
```diff
@@ -0,0 +1,29 @@
+ $ npm test
+
+[IMPORTANT]
+====
```
I forgot to keep js-babel in sync with new changes. Will need to update it

obilodeau replied:
Fixed with 52c4e3f

obilodeau reviewed (changes requested):
New Todo item

obilodeau committed:
Tests: test/js-babel/ now works and highlights the change we made

Also allows us to retest in the future and not forget about that use-case.

Mogztter commented on test/js-babel/main.js:
```diff
@@ -0,0 +1,21 @@
+//import 'asciidoctor-reveal.js'
+
+// NOW SUPPORTED
+import AsciidoctorRevealJs from 'asciidoctor-reveal.js'
```
Just a nitpick but I think the project name is lowercase `reveal.js`, so the variable should be `AsciidoctorRevealjs` ?

obilodeau replied:
sure, why not

ggrossetie reviewed (commented):
I can't approve (my own pull request) but it looks good

obilodeau committed:
consistency: renamed variable to fit upstream's case
